src/openms/source/FORMAT/HANDLERS/MzIdentMLHandler.cpp

```diff
@@ -892,10 +894,30 @@ namespace OpenMS
                   else
                   {
                     acc = mod->getPSIMODAccession();
-                    p += "\">\n\t\t\t<cvParam accession=\"XLMOD:" + acc.suffix(':');
-                    p += "\" name=\"" +  mod->getId();
-                    p += "\" cvRef=\"XLMOD\"/>";
-                    p += "\n\t\t</Modification>\n";
+                    if (!acc.empty())
+                    {
+                      p += "\">\n\t\t\t<cvParam accession=\"XLMOD:" + acc.suffix(':');
+                      p += "\" name=\"" +  mod->getId();
+                      p += "\" cvRef=\"XLMOD\"/>";
+                      p += "\n\t\t</Modification>\n";
+                    }
+                    else
+                    {
+                      // We have an unknown modification, so lets write unknown
+                      // and at least try to write down the delta mass.
+                      if (mod->getDiffMonoMass() > 0.0)
```

Comment:
yes, good catch. should we test `!= 0` or should we do `fabs() < 1e-5` where I think the first one *should* be safe, right?